packages/web3-utils/src/converters.ts

```diff
@@ -86,7 +86,8 @@ export const bytesToBuffer = (data: Bytes): Buffer | never => {
 	}
 
 	if (typeof data === 'string' && isHexStrict(data)) {
-		return Buffer.from(data.slice(2), 'hex');
+		const finalData = data.length % 2 === 0 ? data : `0${data}`;
```

Comment:
```suggestion
		const finalData = data.replace(/^(?!0x)/,'0x') ;
```

Reply:
It’s just stating if any starting does not start with 0x, just replace the start with 0x.
Also, add a few test cases for this. It’s very tricky but essential code.

Reply:
I could use this function in preprocessing of `data` variable to ensure it starts with `0x`, but, what I want to do is to ensure that the length of the string is even, in order to `Buffer.from` work as wanted.